src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+import static java.lang.System.arraycopy;
+
 public class ArrayList<T> implements List<T> {
+    private static final double SIZE_MULTIPLIER = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size = 0;
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            elements = (T[]) new Object[capacity];
+        } else if (capacity == 0) {
+            elements = (T[]) new Object[DEFAULT_CAPACITY];
+        } else {
+            throw new IllegalArgumentException("Invalid capacity");
+        }
+    }
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfFull();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+        } else {
+            rangeCheck(index);
+            growIfFull();
+            Object[] newData = elements.clone();
+            elements[index] = value;
+            arraycopy(newData, index, elements, index + 1, newData.length - index - 1);
+            size++;
+        }
```

Comment:
No second array array is required.
Should be something like below. Same array is a source and also is a destination of copying.
Probably need to play a bit with indexes, check with tests
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        rangeCheck(index);
        growIfFull();
        arraycopy(elements, index, elements, index + 1, elements.length - index - 1);
        elements[index] = value;
        size++;
```